cpp/test/sg/rf_treelite_test.cu

```diff
@@ -0,0 +1,416 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuda_utils.h>
+#include <gtest/gtest.h>
+#include <test_utils.h>
+#include <treelite/c_api.h>
+#include <treelite/c_api_runtime.h>
+#include <cstdlib>
+#include <fstream>
+#include <iostream>
+#include "decisiontree/decisiontree_impl.h"
+#include "ml_utils.h"
+#include "randomforest/randomforest.hpp"
+
+/** check for system errors and assert accordingly */
+#define SYSTEMCALL_CHECK(call)                                    \
```

Comment:
This is somewhat a misnomer, as `system()` is not a system call, but a libc call. Feel free to rename this macro, or drop it and just use `ASSERT()`, as it is only used twice.